src/adapters/methods/reft.py

```diff
@@ -66,9 +66,15 @@ def __init__(self, in_features: int, config: ReftConfig):
 
     def _gather_adapted_states(self, hidden_states: torch.Tensor):
         context = ForwardContext.get_context()
-        bsz, _, ddim = hidden_states.size()
+        bsz, seq_len, ddim = hidden_states.size()
+
+        # if cached indexing matrices are computed for different hidden_states size -> recompute
+        cache_invalidated = False
+        if hasattr(context, "pref_idx") and hasattr(context, "suff_idx"):
+            cache_invalidated = context.suff_idx.size(1) != seq_len
```

Comment:
great catch! should we check the full shape here since bsz and ddim might also change?

Reply:
Yes, then we have all potential cases covered 👍

Reply:
However I now realized that my checking logic was not correct; the indixing matrices and hidden_states do never have the same value at dim1:
- the hidden_states[1] represent the sequence length
- the suff_idx[1] represents the number of positions
We need to check for the actual values of suff_idx to see if the indexing values are out of bounds. I adapted the logic and when the tests pass locally I will push the changes for review